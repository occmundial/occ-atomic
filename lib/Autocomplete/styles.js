import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
