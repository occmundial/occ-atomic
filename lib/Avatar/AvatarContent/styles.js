import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
