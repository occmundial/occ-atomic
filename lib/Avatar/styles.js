import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
