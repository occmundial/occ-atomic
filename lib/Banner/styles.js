import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
