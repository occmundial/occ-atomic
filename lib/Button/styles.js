import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
