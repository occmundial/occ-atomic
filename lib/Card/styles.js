import colors from '../subatomic/colors';
import shadows from '../subatomic/shadows';
import spacing from '../subatomic/spacing';
import grid from '../subatomic/grid';

const styles = {
    card: {
        background: colors.white,
        borderRadius: 6,
        padding: spacing.small,
        position: 'relative',
        transition: '0.3s box-shadow, 0.3s transform',
        [`@media screen and (min-width:${grid.sm}px)`]: {
            padding: spacing.medium
        }
    },
    cardNoPadding: {
        extend: 'card',
        padding: 0,
        [`@media screen and (min-width:${grid.sm}px)`]: {
            padding: 0
        }
    },
    flat: {
        boxShadow: shadows.lvl1,
        zIndex: 'auto'
    },
    rest: {
        boxShadow: shadows.lvl0,
        zIndex: 'auto'
    },
    raised: {
        boxShadow: shadows.lvl4,
        zIndex: 4
    },
    raisable: {
        '&:hover': {
            boxShadow: shadows.lvl4,
            transform: `translateY(-${spacing.xTiny}px)`,
            zIndex: 4
        }
    }
};

export default styles;
