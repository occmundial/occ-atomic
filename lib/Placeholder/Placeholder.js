import React from 'react';
import PropTypes from 'prop-types';
import Flexbox from '../Flexbox';

/** Use this component to simulate the layout of a content that is still loading. You can easily match the real layout using multiple placeholders that match the text and grid. */
const Placeholder = ({ classes, textSize, round, top, bottom }) => (
    <Flexbox display="flex" alignItems="center" className={`${textSize ? classes[textSize] : ''}${top ? ` ${classes[`top${top}`]}` : ''}${bottom ? ` ${classes[`bottom${bottom}`]}` : ''}`}>
        <div className={`${classes.placeholder}${round ? ` ${classes.round}` : ''}`} />
    </Flexbox>
);

Placeholder.propTypes = {
    classes: PropTypes.object.isRequired,
    /** Simulates a row of text. It has the same size and line-height as the Text component. */
    textSize: PropTypes.oneOf([
        'hero', 'headline', 'heading', 'subheading', 'large', 'standard', 'small', 'micro'
    ]),
    /** Spacing in the top of the placeholder. */
    top: PropTypes.oneOf([
        'xTiny', 'tiny', 'small', 'base', 'medium', 'large', 'xLarge'
    ]),
    /** Spacing in the bottom of the placeholder. */
    bottom: PropTypes.oneOf([
        'xTiny', 'tiny', 'small', 'base', 'medium', 'large', 'xLarge'
    ]),
    /** Width of the placeholder. You can use numbers or a string with a percentage. */
    width: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    /** Height of the placeholder. You can use numbers or a string with a percentage. */
    height: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
    ]),
    /** It applies a border radius of 50%. Use it with width and height to create a circle to simulate an avatar or profile picture. */
    round: PropTypes.bool,
    /** It applies gradient left background */
    start: PropTypes.bool
};

export default Placeholder;
