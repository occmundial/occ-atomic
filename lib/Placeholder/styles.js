import spacing from '../subatomic/spacing';

export default {
    '@keyframes progress': {
        '0%': {
            transform: 'translate3d(-100%, 0, 0)'
        },
        '100%': {
            transform: 'translate3d(100%, 0, 0)'
        }
    },
    placeholder: {
        width: props => props.width ? props.width : '100%',
        height: props => props.height ? props.height : 14,
        borderRadius: spacing.xTiny,
        position: 'relative',
        overflow: 'hidden',
        background: '#f2f2f2',
        mixBlendMode: 'multiply',
        '&:after': {
            content: '""',
            position: 'absolute',
            top: 0,
            left: 0,
            width: '100%',
            height: '100%',
            background: `linear-gradient(90deg, rgba(204, 204, 204, 0), rgba(204, 204, 204, 0.7), rgba(204, 204, 204, 0))`,
            animation: 'progress 2s ease-in-out infinite'
        },
    },
    round: {
        borderRadius: '50%'
    },

    hero: {
        height: 72,
        '& $placeholder': {
            height: 57
        }
    },
    headline: {
        height: 48,
        '& $placeholder': {
            height: 36
        }
    },
    heading: {
        height: 32,
        '& $placeholder': {
            height: 22
        }
    },
    subheading: {
        height: 24,
        '& $placeholder': {
            height: 17
        }
    },
    large: {
        height: 24,
        '& $placeholder': {
            height: 17
        }
    },
    standard: {
        height: 24,
        '& $placeholder': {
            height: 15
        }
    },
    small: {
        height: 16,
        '& $placeholder': {
            height: 13
        }
    },
    micro: {
        height: 16,
        '& $placeholder': {
            height: 11
        }
    },
    topxTiny: {
        marginTop: spacing.xTiny
    },
    toptiny: {
        marginTop: spacing.tiny
    },
    topsmall: {
        marginTop: spacing.small
    },
    topbase: {
        marginTop: spacing.base
    },
    topmedium: {
        marginTop: spacing.medium
    },
    toplarge: {
        marginTop: spacing.large
    },
    topxLarge: {
        marginTop: spacing.xLarge
    },
    bottomxTiny: {
        marginBottom: spacing.xTiny
    },
    bottomtiny: {
        marginBottom: spacing.tiny
    },
    bottomsmall: {
        marginBottom: spacing.small
    },
    bottombase: {
        marginBottom: spacing.base
    },
    bottommedium: {
        marginBottom: spacing.medium
    },
    bottomlarge: {
        marginBottom: spacing.large
    },
    bottomxLarge: {
        marginBottom: spacing.xLarge
    }
};
