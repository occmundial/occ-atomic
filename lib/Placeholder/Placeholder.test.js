import React from 'react';
import { shallow } from 'enzyme';

import Placeholder from './Placeholder';
import styles from './styles';
import PlaceholderJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });
const classes = Object.keys(styles).reduce(reduceClasses, {});

describe('Placeholder', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<Placeholder classes={classes} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('has the right classNames', () => {
		const wrapper = shallow(<Placeholder classes={classes} />);
		expect(wrapper.prop('className')).toBe('placeholder light  ');
		wrapper.setProps({ textSize: 'headline' });
		expect(wrapper.prop('className')).toBe('placeholder light  h2 ');
		wrapper.setProps({ top: 'tiny' });
		expect(wrapper.prop('className')).toBe('placeholder light  h2 toptiny  ');
		wrapper.setProps({ bottom: 'small' });
		expect(wrapper.prop('className')).toBe('placeholder light  h2 toptiny  bottomsmall ');
	});
});

describe("PlaceholderJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <PlaceholderJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Placeholder styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});
