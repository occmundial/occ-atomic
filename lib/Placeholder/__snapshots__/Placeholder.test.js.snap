// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Placeholder
    classes={
      Object {
        "@keyframes placeholderPulse": "@keyframes placeholderPulse",
        "bottombase": "bottombase",
        "bottomlarge": "bottomlarge",
        "bottommedium": "bottommedium",
        "bottomsmall": "bottomsmall",
        "bottomtiny": "bottomtiny",
        "bottomxLarge": "bottomxLarge",
        "bottomxTiny": "bottomxTiny",
        "heading": "heading",
        "headline": "headline",
        "hero": "hero",
        "large": "large",
        "micro": "micro",
        "placeholder": "placeholder",
        "round": "round",
        "small": "small",
        "standard": "standard",
        "subheading": "subheading",
        "topbase": "topbase",
        "toplarge": "toplarge",
        "topmedium": "topmedium",
        "topsmall": "topsmall",
        "toptiny": "toptiny",
        "topxLarge": "topxLarge",
        "topxTiny": "topxTiny",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": <div
        className="placeholder"
      />,
      "className": "",
      "display": "flex",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "placeholder",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": <div
          className="placeholder"
        />,
        "className": "",
        "display": "flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "placeholder",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Placeholder styles matches the snapshot 1`] = `
Object {
  "@keyframes placeholderPulse": Object {
    "0%": Object {
      "opacity": 0.5,
    },
    "100%": Object {
      "opacity": 0.5,
    },
    "50%": Object {
      "opacity": 1,
    },
  },
  "bottombase": Object {
    "marginBottom": 24,
  },
  "bottomlarge": Object {
    "marginBottom": 48,
  },
  "bottommedium": Object {
    "marginBottom": 32,
  },
  "bottomsmall": Object {
    "marginBottom": 16,
  },
  "bottomtiny": Object {
    "marginBottom": 8,
  },
  "bottomxLarge": Object {
    "marginBottom": 64,
  },
  "bottomxTiny": Object {
    "marginBottom": 4,
  },
  "heading": Object {
    "& $placeholder": Object {
      "height": 22,
    },
    "height": 32,
  },
  "headline": Object {
    "& $placeholder": Object {
      "height": 36,
    },
    "height": 48,
  },
  "hero": Object {
    "& $placeholder": Object {
      "height": 57,
    },
    "height": 72,
  },
  "large": Object {
    "& $placeholder": Object {
      "height": 16,
    },
    "height": 24,
  },
  "micro": Object {
    "& $placeholder": Object {
      "height": 10,
    },
    "height": 16,
  },
  "placeholder": Object {
    "animation": "placeholderPulse 1s infinite ease-in-out",
    "background": "#f5f5f5",
    "borderRadius": 4,
    "height": [Function],
    "width": [Function],
  },
  "round": Object {
    "borderRadius": "50%",
  },
  "small": Object {
    "& $placeholder": Object {
      "height": 12,
    },
    "height": 16,
  },
  "standard": Object {
    "& $placeholder": Object {
      "height": 14,
    },
    "height": 24,
  },
  "subheading": Object {
    "& $placeholder": Object {
      "height": 16,
    },
    "height": 24,
  },
  "topbase": Object {
    "marginTop": 24,
  },
  "toplarge": Object {
    "marginTop": 48,
  },
  "topmedium": Object {
    "marginTop": 32,
  },
  "topsmall": Object {
    "marginTop": 16,
  },
  "toptiny": Object {
    "marginTop": 8,
  },
  "topxLarge": Object {
    "marginTop": 64,
  },
  "topxTiny": Object {
    "marginTop": 4,
  },
}
`;

exports[`PlaceholderJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Placeholder) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "bottombase": "Placeholder-bottombase-0-1-45 Placeholder-bottombase-0-1-21",
        "bottomlarge": "Placeholder-bottomlarge-0-1-47 Placeholder-bottomlarge-0-1-23",
        "bottommedium": "Placeholder-bottommedium-0-1-46 Placeholder-bottommedium-0-1-22",
        "bottomsmall": "Placeholder-bottomsmall-0-1-44 Placeholder-bottomsmall-0-1-20",
        "bottomtiny": "Placeholder-bottomtiny-0-1-43 Placeholder-bottomtiny-0-1-19",
        "bottomxLarge": "Placeholder-bottomxLarge-0-1-48 Placeholder-bottomxLarge-0-1-24",
        "bottomxTiny": "Placeholder-bottomxTiny-0-1-42 Placeholder-bottomxTiny-0-1-18",
        "heading": "Placeholder-heading-0-1-29 Placeholder-heading-0-1-5",
        "headline": "Placeholder-headline-0-1-28 Placeholder-headline-0-1-4",
        "hero": "Placeholder-hero-0-1-27 Placeholder-hero-0-1-3",
        "large": "Placeholder-large-0-1-31 Placeholder-large-0-1-7",
        "micro": "Placeholder-micro-0-1-34 Placeholder-micro-0-1-10",
        "placeholder": "Placeholder-placeholder-0-1-25 Placeholder-placeholder-0-1-1",
        "round": "Placeholder-round-0-1-26 Placeholder-round-0-1-2",
        "small": "Placeholder-small-0-1-33 Placeholder-small-0-1-9",
        "standard": "Placeholder-standard-0-1-32 Placeholder-standard-0-1-8",
        "subheading": "Placeholder-subheading-0-1-30 Placeholder-subheading-0-1-6",
        "topbase": "Placeholder-topbase-0-1-38 Placeholder-topbase-0-1-14",
        "toplarge": "Placeholder-toplarge-0-1-40 Placeholder-toplarge-0-1-16",
        "topmedium": "Placeholder-topmedium-0-1-39 Placeholder-topmedium-0-1-15",
        "topsmall": "Placeholder-topsmall-0-1-37 Placeholder-topsmall-0-1-13",
        "toptiny": "Placeholder-toptiny-0-1-36 Placeholder-toptiny-0-1-12",
        "topxLarge": "Placeholder-topxLarge-0-1-41 Placeholder-topxLarge-0-1-17",
        "topxTiny": "Placeholder-topxTiny-0-1-35 Placeholder-topxTiny-0-1-11",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "bottombase": "Placeholder-bottombase-0-1-45 Placeholder-bottombase-0-1-21",
          "bottomlarge": "Placeholder-bottomlarge-0-1-47 Placeholder-bottomlarge-0-1-23",
          "bottommedium": "Placeholder-bottommedium-0-1-46 Placeholder-bottommedium-0-1-22",
          "bottomsmall": "Placeholder-bottomsmall-0-1-44 Placeholder-bottomsmall-0-1-20",
          "bottomtiny": "Placeholder-bottomtiny-0-1-43 Placeholder-bottomtiny-0-1-19",
          "bottomxLarge": "Placeholder-bottomxLarge-0-1-48 Placeholder-bottomxLarge-0-1-24",
          "bottomxTiny": "Placeholder-bottomxTiny-0-1-42 Placeholder-bottomxTiny-0-1-18",
          "heading": "Placeholder-heading-0-1-29 Placeholder-heading-0-1-5",
          "headline": "Placeholder-headline-0-1-28 Placeholder-headline-0-1-4",
          "hero": "Placeholder-hero-0-1-27 Placeholder-hero-0-1-3",
          "large": "Placeholder-large-0-1-31 Placeholder-large-0-1-7",
          "micro": "Placeholder-micro-0-1-34 Placeholder-micro-0-1-10",
          "placeholder": "Placeholder-placeholder-0-1-25 Placeholder-placeholder-0-1-1",
          "round": "Placeholder-round-0-1-26 Placeholder-round-0-1-2",
          "small": "Placeholder-small-0-1-33 Placeholder-small-0-1-9",
          "standard": "Placeholder-standard-0-1-32 Placeholder-standard-0-1-8",
          "subheading": "Placeholder-subheading-0-1-30 Placeholder-subheading-0-1-6",
          "topbase": "Placeholder-topbase-0-1-38 Placeholder-topbase-0-1-14",
          "toplarge": "Placeholder-toplarge-0-1-40 Placeholder-toplarge-0-1-16",
          "topmedium": "Placeholder-topmedium-0-1-39 Placeholder-topmedium-0-1-15",
          "topsmall": "Placeholder-topsmall-0-1-37 Placeholder-topsmall-0-1-13",
          "toptiny": "Placeholder-toptiny-0-1-36 Placeholder-toptiny-0-1-12",
          "topxLarge": "Placeholder-topxLarge-0-1-41 Placeholder-topxLarge-0-1-17",
          "topxTiny": "Placeholder-topxTiny-0-1-35 Placeholder-topxTiny-0-1-11",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
